tests/integration/testutils/testserver.go

```diff
@@ -56,7 +56,7 @@ func StartAuthenticatorTestFramework(t *testing.T, stopCh <-chan struct{}, setup
 		t.Fatal(err)
 	}
 
-	adminClient, kubeAPIServerClientConfig := framework.StartTestServer(t, stopCh, framework.TestServerSetup{
+	adminClient, kubeAPIServerClientConfig, tearDownFn := framework.StartTestServer(t, framework.TestServerSetup{
```

Comment:
`TestServerSetup` now returns a tear down function we need to call for cleaning up

Reply:
Definitely more user-friendly.